src/main/java/net/dv8tion/jda/core/utils/WidgetUtil.java

```diff
@@ -0,0 +1,685 @@
+import net.dv8tion.jda.core.entities.Guild;
+import net.dv8tion.jda.core.entities.IMentionable;
+import net.dv8tion.jda.core.entities.ISnowflake;
+import net.dv8tion.jda.core.entities.impl.UserImpl;
+import net.dv8tion.jda.core.utils.WidgetUtil.Widget.Channel;
+import net.dv8tion.jda.core.utils.WidgetUtil.Widget.Member;
+import org.apache.http.util.Args;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+/**
+ * The WidgetUtil is a class for interacting with various facets of Discord's
+ * guild widgets
+ */
+public class WidgetUtil {
+    
+    public static final String WIDGET_PNG = "https://discordapp.com/api/guilds/%s/widget.png?style=%s";
+    public static final String WIDGET_URL = "https://discordapp.com/api/guilds/%s/widget.json";
+    public static final String WIDGET_HTML = "<iframe src=\"https://discordapp.com/widget?id=%s&theme=%s\" width=\"%d\" height=\"%d\" allowtransparency=\"true\" frameborder=\"0\"></iframe>";
+    
+    /**
+     * Gets the banner image for the specified guild of the specified type.
+     * This banner will only be available if the guild in question has the
+     * Widget enabled.
+     * 
+     * @param guild The guild
+     * @param type The type (visual style) of the banner
+     * @return A String containing the URL of the banner image
+     */
+    public static String getWidgetBanner(Guild guild, BannerType type)
+    {
+        Args.notNull(guild, "Guild");
+        return getWidgetBanner(guild.getId(), type);
+    }
+    
+    /**
+     * Gets the banner image for the specified guild of the specified type.
+     * This banner will only be available if the guild in question has the
+     * Widget enabled. Additionally, this method can be used independently of
+     * being on the guild in question.
+     * 
+     * @param guildId the guild ID
+     * @param type The type (visual style) of the banner
+     * @return A String containing the URL of the banner image
+     */
+    public static String getWidgetBanner(String guildId, BannerType type)
+    {
+        Args.notNull(guildId, "GuildId");
+        Args.notNull(type, "BannerType");
+        return String.format(WIDGET_PNG, guildId, type.name().toLowerCase());
+    }
+    
+    /**
+     * Represents the available banner types
+     * Each of these has a different appearance:
+     * 
+     * Shield - tiny, only contains Discord logo and online count
+     * 
+     * Banner1 - medium, contains server name, icon, and online count, and a 
+     * "Powered by Discord" bar on the bottom
+     * 
+     * Banner2 - small, contains server name, icon, and online count, and a 
+     * Discord logo on the side
+     * 
+     * Banner3 - medium, contains server name, icon, and online count, and a 
+     * Discord logo with a "Chat Now" bar on the bottom
+     * 
+     * Banner4 - large, contains a very big Discord logo, server name, icon, and 
+     * online count, and a big "Join My Server" button
+     */
+    public enum BannerType
```

Comment:
While the javadoc is pretty to look at in source, it will not appear this way when javadoc html is generated. You need formatting tags. Preview it in IDE to make sure it is properly formatted. 

Note: Consider using **bold** for the label names (Sheild, Banner1, etc)